src/client/Dentry.h

```diff
@@ -84,7 +84,7 @@ class Dentry : public LRUObject {
     if (dir) {
       ret = dir->parent_inode->make_path_string(s);
     } else {
-      s = "???";
+      return false;
```

Comment:
This seems like a better approach — if we can't link all the way to our mount point, say so explicitly.

But I think you're over-specifying the conditions this could happen under, in your commit description. Right now, we're only invoking ```make_path_string()``` in ```setattr()``` and ```open()```, but I'm not sure there's anything inherent to those functions limiting it to them, is there?

Reply:
One method of definitely ending up in this case is if an NFS client does a lookup_by_ino, and then issues an open, btw — that inode is quite probably not linked up. (This surprised me on discovery, but IIUC is definitely the case.)

Reply:
> This seems like a better approach — if we can't link all the way to our mount point, say so explicitly.
> 
> But I think you're over-specifying the conditions this could happen under, in your commit description. Right now, we're only invoking `make_path_string()` in `setattr()` and `open()`, but I'm not sure there's anything inherent to those functions limiting it to them, is there?

Currently I think there is no. I will improve the commit comment.



> One method of definitely ending up in this case is if an NFS client does a lookup_by_ino, and then issues an open, btw — that inode is quite probably not linked up. (This surprised me on discovery, but IIUC is definitely the case.)

Okay, sounds reasonable.